src/main/java/net/minecraftforge/items/wrapper/ShulkerItemStackInvWrapper.java

```diff
@@ -0,0 +1,247 @@
+            }
+        }
+        else
+        {
+            if (!this.isItemValid(slot, stack))
+                return stack;
+
+            m = Math.min(stack.getMaxStackSize(), getSlotLimit(slot));
+            if (m < stack.getCount())
+            {
+                // copy the stack to not modify the original one
+                stack = stack.copy();
+                if (!simulate)
+                {
+                    itemStacks.set(slot, stack.split(m));
+                    this.setItemList(itemStacks);
+                    return stack;
+                }
+                else
+                {
+                    stack.shrink(m);
+                    return stack;
+                }
+            }
+            else
+            {
+                if (!simulate)
+                {
+                    itemStacks.set(slot, stack);
+                    this.setItemList(itemStacks);
+                }
+                return ItemStack.EMPTY;
+            }
+        }
+    }
+
+    @Override
+    @NotNull
+    public ItemStack extractItem(int slot, int amount, boolean simulate)
+    {
+        NonNullList<ItemStack> itemStacks = getItemList();
+        if (amount == 0)
+            return ItemStack.EMPTY;
+
+        ItemStack stackInSlot = itemStacks.get(slot);
+
+        if (stackInSlot.isEmpty())
+            return ItemStack.EMPTY;
+
+        if (simulate)
+        {
+            if (stackInSlot.getCount() < amount)
+            {
+                return stackInSlot.copy();
+            }
+            else
+            {
+                ItemStack copy = stackInSlot.copy();
+                copy.setCount(amount);
+                return copy;
+            }
+        }
+        else
+        {
+            int m = Math.min(stackInSlot.getCount(), amount);
+
+            ItemStack decrStackSize = ContainerHelper.removeItem(itemStacks, slot, m);
+            this.setItemList(itemStacks);
+            return decrStackSize;
+        }
+    }
+
+    @Override
+    public int getSlotLimit(int slot)
+    {
+        return 64;
+    }
+
+    @Override
+    public boolean isItemValid(int slot, @NotNull ItemStack stack)
+    {
+        return stack.getItem().canFitInsideContainerItems();
+    }
+
+    public NonNullList<ItemStack> getItemList()
+    {
+        NonNullList<ItemStack> itemStacks = NonNullList.withSize(getSlots(), ItemStack.EMPTY);
+        CompoundTag rootTag = BlockItem.getBlockEntityData(this.stack);
+        if (rootTag != null && rootTag.contains("Items", 9)) {
+            ContainerHelper.loadAllItems(rootTag, itemStacks);
+        }
+        return itemStacks;
+    }
+
+    public void setItemList(NonNullList<ItemStack> itemStacks)
+    {
+        CompoundTag existing = BlockItem.getBlockEntityData(this.stack);
+        BlockItem.setBlockEntityData(this.stack, BlockEntityType.SHULKER_BOX,
+                ContainerHelper.saveAllItems(existing == null ? new CompoundTag() : existing, itemStacks));
+    }
```

Comment:
This should really use a cache and only deserialize when the stack's NBT tag has not changed.